clientconn_parsed_target_test.go

```diff
@@ -0,0 +1,157 @@
+		{target: "unix-abstract:unix:///abc", wantParsed: resolver.Target{Scheme: "unix-abstract", Authority: "", Endpoint: "unix:///abc"}},
+		{target: "unix-abstract:///a/b/c", wantParsed: resolver.Target{Scheme: "unix-abstract", Authority: "", Endpoint: "/a/b/c"}},
+		{target: "unix-abstract:///", wantParsed: resolver.Target{Scheme: "unix-abstract", Authority: "", Endpoint: "/"}},
+		{target: "unix-abstract://authority", wantParsed: resolver.Target{Scheme: "unix-abstract", Authority: "", Endpoint: "//authority"}},
+		{target: "passthrough:///unix:///a/b/c", wantParsed: resolver.Target{Scheme: "passthrough", Authority: "", Endpoint: "unix:///a/b/c"}},
+
+		// If we can only parse part of the target.
+		{target: "://", wantParsed: resolver.Target{Scheme: defScheme, Authority: "", Endpoint: "://"}},
+		{target: "unix://domain", wantParsed: resolver.Target{Scheme: defScheme, Authority: "", Endpoint: "unix://domain"}},
+	}
+
+	for _, test := range tests {
+		t.Run(test.target, func(t *testing.T) {
+			cc, err := Dial(test.target, WithInsecure())
+			if err != nil {
+				t.Fatalf("Dial(%q) failed: %v", test.target, err)
+			}
+			defer cc.Close()
+
+			if gotParsed := cc.parsedTarget; gotParsed != test.wantParsed {
+				t.Errorf("cc.parsedTarget = %+v, want %+v", gotParsed, test.wantParsed)
+			}
+		})
+	}
+}
+
+func (s) TestParsedTarget_Failure_WithoutCustomDialer(t *testing.T) {
+	targets := []string{
+		"unix://a/b/c",
+		"unix-abstract://authority/a/b/c",
+	}
+
+	for _, target := range targets {
+		t.Run(target, func(t *testing.T) {
+			if cc, err := Dial(target, WithInsecure()); err == nil {
+				defer cc.Close()
+				t.Fatalf("Dial(%q) succeeded cc.parsedTarget = %+v, expected to fail", target, cc.parsedTarget)
+			}
+		})
+	}
+}
+
+func (s) TestParsedTarget_WithCustomDialer(t *testing.T) {
+	defScheme := resolver.GetDefaultScheme()
+	tests := []struct {
+		target            string
+		wantParsed        resolver.Target
+		wantDialerAddress string
+	}{
+		// unix:[local_path], unix:[/absolute], and unix://[/absolute] have
+		// different behaviors with a custom dialer.
+		{
+			target:            "unix:a/b/c",
+			wantParsed:        resolver.Target{Scheme: defScheme, Authority: "", Endpoint: "unix:a/b/c"},
+			wantDialerAddress: "unix:a/b/c",
+		},
+		{
+			target:            "unix:/a/b/c",
+			wantParsed:        resolver.Target{Scheme: defScheme, Authority: "", Endpoint: "unix:/a/b/c"},
+			wantDialerAddress: "unix:/a/b/c",
+		},
+		{
+			target:            "unix:///a/b/c",
+			wantParsed:        resolver.Target{Scheme: "unix", Authority: "", Endpoint: "/a/b/c"},
+			wantDialerAddress: "unix:///a/b/c",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.target, func(t *testing.T) {
+			dialer := func(ctx context.Context, address string) (net.Conn, error) {
+				if address != test.wantDialerAddress {
+					return nil, fmt.Errorf("address in custom dialer is %q, want %q", address, test.wantDialerAddress)
```

Comment:
Should this be `t.Errorf`?

Or should we be pushing `address` onto a channel, returning an unconditional error, then validating the address on the channel in the main test goroutine?

Reply:
Done.